src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -316,6 +338,26 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  private void recallItem(InnReachTransaction transaction, UUID itemId) {
+    List<RequestDTO> requestList = requestService.getRequestsByItemId(itemId);
+    for (RequestDTO request : requestList) {
+      if (request.getStatus() == OPEN_NOT_YET_FILLED) {
+        transaction.setState(RECALL);
+        var loan = loanService.getById(transaction.getHold().getFolioLoanId());
+        var loanDueDate = loan.getDueDate().toInstant().truncatedTo(ChronoUnit.SECONDS);
+        var loanIntegerDueDate = (int) (loanDueDate.getEpochSecond());
+        transaction.getHold().setDueDateTime(loanIntegerDueDate);
+        notifier.reportRecallRequested(transaction, loanDueDate);
+      } else if (request.getStatus() != OPEN_NOT_YET_FILLED || request.getStatus() != OPEN_AWAITING_PICKUP
+        || request.getStatus() != OPEN_IN_TRANSIT || request.getStatus() != OPEN_AWAITING_DELIVERY) {
+        var centralServerId = centralServerService.getCentralServerIdByCentralCode(transaction.getCentralServerCode());
+        var recallUserDTO = recallUserService.getInnReachRecallUser(centralServerId);
+        var recallUser = recallUserMapper.toEntity(recallUserDTO);
+        eventPublisher.publishEvent(RecallRequestEvent.of(transaction.getHold(), recallUser));
```

Comment:
adjust static constructor of RecallRequestEvent instead of applying  recallUserMapper